share/new_eds/rsmt2d.go

```diff
@@ -0,0 +1,116 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+var _ Accessor = Rsmt2D{}
+
+// Rsmt2D is a rsmt2d based in-memory implementation of Accessor.
+type Rsmt2D struct {
+	*rsmt2d.ExtendedDataSquare
+}
+
+// Close does nothing.
+func (eds Rsmt2D) Close() error {
+	return nil
+}
+
+// Size returns the size of the Extended Data Square.
+func (eds Rsmt2D) Size() int {
+	return int(eds.Width())
+}
+
+// Sample returns share and corresponding proof for row and column indices.
+func (eds Rsmt2D) Sample(
+	_ context.Context,
+	rowIdx, colIdx int,
+) (shwap.Sample, error) {
+	return eds.SampleForProofAxis(rowIdx, colIdx, rsmt2d.Row)
+}
+
+// SampleForProofAxis samples a share from an Extended Data Square based on the provided
+// row and column indices and proof axis. It returns a sample with the share and proof.
+func (eds Rsmt2D) SampleForProofAxis(
+	rowIdx, colIdx int,
+	proofType rsmt2d.Axis,
+) (shwap.Sample, error) {
+	var axisIdx, shrIdx int
+	switch proofType {
+	case rsmt2d.Row:
+		axisIdx, shrIdx = rowIdx, colIdx
+	case rsmt2d.Col:
+		axisIdx, shrIdx = colIdx, rowIdx
+	default:
+		return shwap.Sample{}, fmt.Errorf("invalid proof type: %d", proofType)
+	}
```

Comment:
this little switcher seems like something nice t have in it's own little func and test for, subtle but small like a nice place for a bug to hide

Reply:
extracted in https://github.com/celestiaorg/celestia-node/pull/3452/commits/8cbd6b461597a0b4ea0a6e8f0e500f78018d6309